src/components/Identicon/index.tsx

```diff
@@ -1,64 +1,24 @@
 import * as React from 'react'
-import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
 
-import { toDataUrl } from './blockies'
+import makeBlockie from 'ethereum-blockies-base64'
+import styled from 'styled-components'
 
-export default class Identicon extends React.PureComponent<any> {
-  private identicon: React.RefObject<HTMLImageElement>
-
-  static defaultProps = {
-    className: '',
-  }
-
-  constructor(props) {
-    super(props)
-
-    this.identicon = React.createRef<HTMLImageElement>()
-  }
-
-  componentDidMount = () => {
-    const { address, diameter } = this.props
-    const image = this.generateBlockieIdenticon(address, diameter)
-    if (this.identicon.current) {
-      this.identicon.current.appendChild(image)
-    }
-  }
-
-  componentDidUpdate = () => {
-    const { address, diameter } = this.props
-    const image = this.generateBlockieIdenticon(address, diameter)
-
-    if (!this.identicon.current) {
-      return
-    }
-
-    const { children } = this.identicon.current
-    for (let i = 0; i < children.length; i += 1) {
-      this.identicon.current.removeChild(children[i])
-    }
-
-    this.identicon.current.appendChild(image)
-  }
-
-  getStyleFrom = (diameter) => ({
-    width: diameter,
-    height: diameter,
-  })
-
-  generateBlockieIdenticon = (address, diameter) => {
-    const image = new window.Image()
-    image.src = toDataUrl(address || ZERO_ADDRESS)
-    image.height = diameter
-    image.width = diameter
-    image.style.borderRadius = `${diameter / 2}px`
+type Props = {
+  address: string
+  diameter: number
+  className?: string
+}
 
-    return image
-  }
+const StyledImg = styled.img<{ diameter: number }>`
+  height: ${({ diameter }) => (diameter ? diameter : 32)}px;
```

Comment:
maybe falling back to default value in two places is too much.. but who knows 😄

Reply:
Done!